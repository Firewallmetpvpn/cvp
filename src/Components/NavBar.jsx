import React from "react";
import { NavLink } from "react-router-dom";

const NavBar = () => {
  return (
    <div
      className="navbar navbar-expand-lg navbar-light bg-light"
      role="navigation"
      aria-current="page"
      style={{ backgroundColor: "#e3f2fd" }}
    >
      <div className="container-fluid">
        <button
          class="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarNav"
          aria-controls="navbarNav"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse" id="navbarNav">
          <ul class="navbar-nav">
            <li className="nav-item">
              <NavLink className="nav-link" to="/">
                Home
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink className="nav-link" to="/about">
                About
              </NavLink>
            </li>
            {/* <li className="nav-item">
              <NavLink className="nav-link" to="/contact">
                Contact
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink className="nav-link" to="/volunteer">
                Volunteer
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink className="nav-link" to="/resources">
                Resources
              </NavLink>
            </li> */}
          </ul>
        </div>
      </div>
    </div>
  );
};

export default NavBar;
