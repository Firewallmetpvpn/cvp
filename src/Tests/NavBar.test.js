import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter } from 'react-router-dom';
import NavBar from '../Components/NavBar';

describe("NavBar component", () => {
    //smoke test: Why - Build Verification
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BrowserRouter> <NavBar/> </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    /*snapshot test: Why - You can see if our component is rendering correctly
    Any changes to to the Dom of NavBar will result in a failed snapshot test.
    We just need to press "w" to activate watch mode then press "u" to update the snapshot.
    our snap shot file will be automatically updated with the new snapshot and will pass our tests*/
    it('renders NavBar by default', () => {
        const wrapper = shallow(<NavBar />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
})