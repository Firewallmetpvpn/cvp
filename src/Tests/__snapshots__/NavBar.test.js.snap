// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders NavBar by default 1`] = `
<div
  aria-current="page"
  className="navbar navbar-expand-lg navbar-light bg-light"
  role="navigation"
  style={
    Object {
      "backgroundColor": "#e3f2fd",
    }
  }
>
  <div
    className="container-fluid"
  >
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      data-bs-target="#navbarNav"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse"
      id="navbarNav"
    >
      <ul
        class="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <NavLink
            className="nav-link"
            to="/"
          >
            Home
          </NavLink>
        </li>
        <li
          className="nav-item"
        >
          <NavLink
            className="nav-link"
            to="/about"
          >
            About
          </NavLink>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
